test/app/core/managers/location_services_test.dart

```diff
@@ -0,0 +1,216 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:geolocator/geolocator.dart';
+import 'package:penhas/app/core/entities/user_location.dart';
+import 'package:penhas/app/core/managers/location_services.dart';
+import 'package:penhas/app/core/states/location_permission_state.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:permission_handler_platform_interface/permission_handler_platform_interface.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+void main() {
+  late LocationServices locationServices;
+
+  setUp(() {
+    GeolocatorPlatform.instance = MockGeolocatorPlatform();
+    PermissionHandlerPlatform.instance = MockPermissionHandlerPlatform();
+
+    locationServices = LocationServices();
+  });
+
+  group(LocationServices, () {
+    test(
+      'currentLocation return a UserLocationEntity for PermissionStatus.granted',
+      () async {
+        // arrange
+        final mockPermissionHandlerPlatform =
+            PermissionHandlerPlatform.instance;
+        debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+
+        final position = Position(
+            longitude: -0.01,
+            latitude: .01,
+            timestamp: DateTime(2023, 2, 3, 9, 20),
+            accuracy: 0.1,
+            altitude: 100.01,
+            heading: 1.1,
+            speed: 12.0,
+            speedAccuracy: 0.0);
+
+        when((() => mockPermissionHandlerPlatform
+                .checkPermissionStatus(Permission.location)))
+            .thenAnswer((_) async => PermissionStatus.granted);
+
+        // act
+        final result = await locationServices.currentLocation();
+
+        // assert
+        expect(
+          result.fold((l) => null, (r) => r),
```

Comment:
retornando nulo no `left` não pode acabar confundindo, dando uma possível impressão de que pode estar vindo do `right`? Talvez retornar o `left` pode facilitar o debug.
```suggestion
          result.fold((l) => l, (r) => r),
```